tests/virt_autotest/alp_guest_installation.pm

```diff
@@ -0,0 +1,65 @@
+# Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: This file handles the guest installation for ALP.
+#          It supports configuring vm profiles with template.
+#
+# Maintainer: Alice <xlai@suse.com>, or VT squad <qe-virt@suse.de>
+
+package alp_guest_installation;
+
+use base unified_guest_installation;
+use strict;
+use warnings;
+use testapi;
+use Data::Dumper;
+
+sub edit_guest_profile_with_template {
+    my ($self, $guest_name) = @_;
+
+    # Initialize config
+    my %_guest_profile = %{$concurrent_guest_installations::guest_instances_profiles{$guest_name}};
+
+    # Handle guest_boot_settings
+    my $_boot_value = '';
+    $_boot_value = 'uefi' if (get_var('VIRT_UEFI_GUEST_INSTALL', ''));
+    $_boot_value = 'firmware=efi' if (get_var('VIRT_SEV_ES_GUEST_INSTALL', ''));
+    $_guest_profile{guest_boot_settings} =~ s/##guest_boot_settings##/$_boot_value/g;
+
+    # Handle guest_storage_others
+    my $_backing_disk_name = get_var('HDD_1', '');
+    $_backing_disk_name =~ /([^\/]+)\.([^\/\.]+)$/m;
+    my $_new_back_name = "$1-back.$2";
+    my $_back_format = $2;
+    $_guest_profile{guest_storage_others} =~ s/##backing_disk_name##/$_new_back_name/g;
+    $_guest_profile{guest_storage_others} =~ s/##backing_format##/$_back_format/g;
+
+    # Handle version
+    my $_version = get_var('VERSION', '');
```

Comment:
I think  ```VERSION``` is test suite level setting which indicates the product under development, unless version of guest here is always equal to ```VERSION```.

Reply:
I get your concern point.  Actually not only VERSION/BUILD, but also `guest_storage_others` won't work for non-developing alp guest atm.

From the motivation,  this dynamic guest profile generation(edit_guest_profile_with_template) is designed for *developing* alp guest installation currently, not for released ones. For released ones, they have to use non-template way, just as I did for sles in validation jobs. It will be good to add a comment for this to let users be aware. Will do it later.
If alp is released and grow to be sth like current sles which has large test matrix for previous alp, it will become a real issue to save released alp guests' similar profiles. At that time, it will be much clearer  what else needs to be considered. Let's then do it together in a new ticket to save rework iterations when alp develops. It will be easy extendable based on current design.  HDYT?

Reply:
I do not think use ```VERSION``` and ```BUILD``` directly is good in any case unless it is absolutely necessary. This usage should be avoided at the beginning instead of refactor. If you want to extend this, then extend in elegant way instead of introducing something that is destined to be refactored.  Actually this is not difficult to achieve, you can give it in guest profile and just retrieve it or you can use test suite level setting to handle this.

Reply:
I do not get your point. Would you please explain further? What would you recommend to do?

Reply:
Do not use ```VERSION``` and ```BUILD``` for guest directly.  Actually this is not difficult to achieve, you can give it in guest profile and just retrieve it or you can use test suite level setting to handle this.

Reply:
I am OK as long as these settings can reflect actual situation.

Reply:
One more thing is your alp guest installation with these new settings should support multiple guest concurrent installation, because current automation underneath already supports this and your extension better support this as well.

Reply:
> One more thing is your alp guest installation with these new settings should support multiple guest concurrent installation, because current automation underneath already supports this and your extension better support this as well.

Yes, it does. You can see it from validation jobs too.

Reply:
Done. Please check it out.